abnf/odata-aggregation-testcases.yaml

```diff
@@ -402,6 +419,18 @@ TestCases:
     Expect:
       - aggrPathPrefix:Sales
 
+  - Name: aggregate - custom aggregate reached via type cast
+    Rule: queryOptions
+    Input: $apply=aggregate(Products/Self.DigitalProduct/Forecast)
+    Expect:
+      - aggrPathPrefix:Products/Self.DigitalProduct
+
+  - Name: aggregate - custom aggregate reached via type cast only
+    Rule: queryOptions
+    Input: $apply=aggregate(Self.DigitalProduct/Forecast)
```

Comment:
seems strange that this is supported but 353 is not supported.  i would have expected that we would require something like $it.